internal/internal_worker_test.go

```diff
@@ -1089,11 +1089,13 @@ func (s *internalWorkerTestSuite) testWorkflowTaskHandlerHelper(params workerExe
 }
 
 func (s *internalWorkerTestSuite) TestWorkflowTaskHandlerWithDataConverter() {
+	cache := getWorkerCache()
 	params := workerExecutionParameters{
 		Namespace:     testNamespace,
 		Identity:      "identity",
 		Logger:        getLogger(),
 		DataConverter: iconverter.NewTestDataConverter(),
+		cache:         cache,
```

Comment:
I couldn't find the test that validates that cache is deleted after all workers are closed.

Reply:
@mfateev Yeah, could use a pointer on that one. The problem is such a test needs to run independently of other tests, since otherwise concurrently running tests will mess up the worker count. I couldn't find a good way to say "run this test independently" without actually changing how we call `go test` -- any ideas there?

Reply:
Fixed this by using another layer of indirection (always the solution :smile: ) to make things more testable.